### ISSUE TITLE
Google Analytics is not tracking anything on my forem server

### ISSUE THREAD

**Issue by danicuki** [forem/forem#16552]
**Describe the bug**
I setup my forem server to user track users with Google Analytics, but when I go to the GA website, nothing is being tracked. 

This is my Google Analytics Tracking ID:

G-1GGLWVRFPS

**To Reproduce**

Just configured the Google Analytics Tracking ID on the admin page (`admin/customization/config`) and click save.

**Expected behavior**

Have all the user access and behavior tracked.

**Screenshots**
![Screen Shot 2022-02-12 at 20 22 40](https://user-images.githubusercontent.com/77552/153731975-7f7f1141-1f3f-4aae-82b4-927cc9fef0a5.png)
![Screen Shot 2022-02-12 at 20 22 52](https://user-images.githubusercontent.com/77552/153731977-936c01bf-0571-4306-8922-77e97a834fd3.png)

**Additional context**
Running on Heroku Server.
online Server: https://www.web3dev.com.br

**Comment by fdocr** [comment]
Thanks for the report @danicuki 

From a quick look at the code it seems like the [Tracking ID is rendered](https://github.com/forem/forem/blob/main/app/views/layouts/application.html.erb#L56) within a [cache directive](https://github.com/forem/forem/blob/main/app/views/layouts/application.html.erb#L47). The cache key uses `user_signed_in?` so my first thought would be to suggest you to sign out and sign in again, to see if that changes anything from your end.

Can you confirm if this works or doesn't work?

<img width="1014" alt="ga-tracking" src="https://user-images.githubusercontent.com/6045239/153915789-4aed9cde-f74b-42de-bfcf-bfa566e9c207.png">

Looking at the instance on the domain you shared I can see a different Tracking ID. Does your environment has `GA_TRACKING_ID` ENV variable configured? There's a chance it's overwritten or using the old value.

Is this a selfhost instance? Are you able to reboot/redeploy?

**Comment by danicuki** [comment]
Hi @fdocr, thanks for asking. 
Yes, I re-deployed the app, making sure I was using the right GA id in both config and env variables. 
About cache, it is not the case, because I don't see any analytics report, even from unlogged users. I suppose that analytics should work for any website traffic, right? I don't get any report at all. The analytics reports are completely empty, and I know we have some traffic on our website.

But I also tried to logout, access without logging in, access after logging in. In any way, no report at all.

**Comment by danicuki** [comment]
I just tested it again and it is working. I noticed that my tests accesses using Brave Browser were not being tracked at all. When I changed to Chrome, the realtime analytics started to show off. In the end, I think what made everything work was to setup the ENV variable correctly. The admin config does not override the ENV, so this should be warned somehow. I will close the issue, but I suggest creating a fix to make the admin config override the env setup.
